catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -77,17 +77,17 @@ public class HadoopCatalogOperations implements CatalogOperations, SupportsSchem
   @VisibleForTesting Optional<Path> catalogStorageLocation;
 
   // For testing only.
-  HadoopCatalogOperations(CatalogEntity entity, EntityStore store) {
-    this.entity = entity;
+  HadoopCatalogOperations(EntityStore store) {
     this.store = store;
   }
 
-  public HadoopCatalogOperations(CatalogEntity entity) {
-    this(entity, GravitinoEnv.getInstance().entityStore());
+  public HadoopCatalogOperations() {
+    this(GravitinoEnv.getInstance().entityStore());
   }
 
   @Override
-  public void initialize(Map<String, String> config) throws RuntimeException {
+  public void initialize(Map<String, String> config, CatalogEntity entity) throws RuntimeException {
```

Comment:
Could the `CatalogOperations` interface add `setCatalogEntity` method?

Reply:
this may make CatalogOperations API more complicated, @jerryshao what do you think?

Reply:
What's the purpose of this change?

Reply:
`CatalogEntity` is passed to every specific XXCatalogOperation in class constructor, like `HiveCatalogOperations(CatalogEntity entity)`, but there is no API interface to define it, how to pass `CatalogEntity` to a custom catalog operation? I propose to pass though initialize interface explicitly.

Reply:
Things doesn't change even if you maven this to `initialize`, `CatalogEntity` is still not a public API we defined currently.

Reply:
do you have some suggestions?

Reply:
I don't have suggestions, I just want to know the purpose of this change, this seems an unrelated change. If there's a benefit I think it is OK to change. If not, then not necessary to change it.

Reply:
The main purpose of moving passing `CatalogEntity` to initialize is to make construct a custom catalog operation using reflect easily.

Reply:
OK, I see.